pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -372,100 +367,80 @@ pub async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
     };
     debug!(?usage_planned, "usage planned");
 
-    // phase2: evict victims batched by timeline
+    // phase2: evict layers
 
     let mut js = tokio::task::JoinSet::new();
+    let limit = 1000;
+
+    let mut evicted = candidates.into_iter().take(evicted_amount).fuse();
+    let mut consumed_all = false;
+
+    // After the evictions, `usage_assumed` is the post-eviction usage,
+    // according to internal accounting.
+    let mut usage_assumed = usage_pre;
+    let mut evictions_failed = LayerCount::default();
+
+    let join_all = async move {
```

Comment:
The `join_all` name confused me on first read. This future is spawning *and* joining.

Reply:
Yeah, I also don't like the name

Reply:
I wonder if I stole it from earlier which was just joining.

Reply:
It's now `evict_layers` per an earlier comment calling this phase "phase 2: evict layers".

Reply:
[d8690dd](https://github.com/neondatabase/neon/pull/6060/commits/d8690ddd7efb7cb3edb76627df2cda3c70e7364d)